File: studyUrl/src/main/java/com/surl/studyurl/global/security/SecurityUser.java
package com.surl.studyurl.global.security;

import lombok.Getter;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;

public class SecurityUser extends User {

    @Getter
    private long id;
    private String oauth_id;  // oauth_id 필드 추가

    public SecurityUser(long id,long oauth_id, String userid, String password, Collection<? extends GrantedAuthority> authorities) {
        super(userid,password,authorities);
        this.id=id;

    }
    public SecurityUser(String userid, String password, boolean enabled, boolean accountNonExpired, boolean credentialsNonExpired, boolean accountNonLocked, Collection<? extends GrantedAuthority> authorities) {
        super(userid, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked, authorities);
        this.id = id;
    }

    public Authentication genAuthentication(){
        Authentication auth = new UsernamePasswordAuthenticationToken(
                this,
                this.getPassword(),
                this.getAuthorities()
        );
        return auth;
    }

}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/global/Security/SecurityUser.kt
package com.krstudy.kapi.global.Security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.core.user.OAuth2User

class SecurityUser(
    val id: Long,
    private val username: String,
    private val password: String,
    private val authorities: Collection<out GrantedAuthority>,
    private val enabled: Boolean = true,
    private val accountNonExpired: Boolean = true,
    private val credentialsNonExpired: Boolean = true,
    private val accountNonLocked: Boolean = true
) : UserDetails, OAuth2User {

    override fun getAuthorities(): Collection<GrantedAuthority> = authorities
    override fun getPassword(): String = password
    override fun getUsername(): String = username
    override fun isAccountNonExpired(): Boolean = accountNonExpired
    override fun isAccountNonLocked(): Boolean = accountNonLocked
    override fun isCredentialsNonExpired(): Boolean = credentialsNonExpired
    override fun isEnabled(): Boolean = enabled

    fun genAuthentication(): Authentication {
        return UsernamePasswordAuthenticationToken(
            this,
            this.password,
            this.authorities
        )
    }

    override fun getAttributes(): Map<String, Any> {
        return emptyMap() // 빈 맵 반환
    }

    override fun getName(): String {
        return username
    }
}
